balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -168,7 +171,9 @@ func init() {
 // which should trigger a transient failure state update.
 func (s) TestWeightedTarget(t *testing.T) {
 	cc := testutils.NewBalancerClientConn(t)
-	wtb := wtbBuilder.Build(cc, balancer.BuildOptions{})
+	wtb := wtbBuilder.Build(cc, balancer.BuildOptions{
+		MetricsRecorder: &stats.NoopMetricsRecorder{},
```

Comment:
This still scares me that we have a required field in this struct, that we added recently...

Reply:
Not in the context of this PR it isn't.  But I think we need to talk about this more as it could be a real problem.